core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/validation/handler/ClusterUpgradeImageValidationHandler.java

```diff
@@ -0,0 +1,68 @@
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade.validation.ClusterUpgradeImageValidationEvent;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade.validation.event.ClusterUpgradeValidationEvent;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade.validation.event.ClusterUpgradeValidationFailureEvent;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade.validation.event.ClusterUpgradeValidationFinishedEvent;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade.validation.event.ClusterUpgradeValidationStateSelectors;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class ClusterUpgradeImageValidationHandler extends ExceptionCatcherEventHandler<ClusterUpgradeImageValidationEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterUpgradeDiskSpaceValidationHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<ClusterUpgradeImageValidationEvent> event) {
+        LOGGER.debug("Accepting cluster upgrade image validation event.");
+        ClusterUpgradeImageValidationEvent request = event.getData();
+        CloudContext cloudContext = request.getCloudContext();
+        Selectable result;
+        try {
+            CloudConnector<Object> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = connector.authentication().authenticate(cloudContext, request.getCloudCredential());
+            for (Validator v : connector.validators(ValidatorType.IMAGE)) {
+                v.validate(ac, request.getCloudStack());
+            }
+            LOGGER.debug("Cluster upgrade image validation succeeded.");
+            result = new ClusterUpgradeValidationEvent(ClusterUpgradeValidationStateSelectors.START_CLUSTER_UPGRADE_DISK_SPACE_VALIDATION_EVENT.selector(),
+                    request.getResourceId(), request.getImageId());
+        } catch (RuntimeException e) {
+            LOGGER.warn("Cluster upgrade image validation failed");
+            result = new ClusterUpgradeValidationFailureEvent(request.getResourceId(), e);
+        }
+        return result;
+    }
+
+    @Override
+    public String selector() {
+        return VALIDATE_IMAGE_EVENT.selector();
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<ClusterUpgradeImageValidationEvent> event) {
+        return new ClusterUpgradeValidationFinishedEvent(resourceId, e);
```

Comment:
is this intentional to use ClusterUpgradeValidationFinishedEvent instead of FailureEvent here and if yes, why?

Reply:
Yes. It's the same for the disk validation step: if there is an error during validation, like azure cannot be reached etc., then let's not stop the upgrade flow. But maybe this validation is simpler than the disk size validator, and probably if this fails then the upgrade will not work, either.
But let's ask @keyki as well, what is your opinion on this?

Reply:
I changed it to failure.